src/common/ufo-scarray.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2017 Karlsruhe Institute of Technology
+ *
+ * This file is part of Ufo.
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ufo-scarray.h"
+
+struct _UfoScarray {
+    GValueArray *array;
+};
+
+UfoScarray *
+ufo_scarray_new (guint n_dims, GType type)
+{
+    UfoScarray *scarray;
+    GValue zero = G_VALUE_INIT;
+    GValue value = G_VALUE_INIT;
+
+    g_value_init (&zero, G_TYPE_INT);
+    g_value_init (&value, type);
+    g_value_set_int (&zero, 0);
+    g_value_transform (&zero, &value);
+
+    scarray = g_new0 (UfoScarray, 1);
+    scarray->array = g_value_array_new (n_dims);
+
+    for (guint i = 0; i < n_dims; i++)
+        g_value_array_append (scarray->array, &value);
+
+    g_value_unset (&zero);
+    g_value_unset (&value);
+
+    return scarray;
+}
+
+void
+ufo_scarray_free (UfoScarray *scarray)
+{
+    g_value_array_free (scarray->array);
+    g_free (scarray);
+}
+
+void
+ufo_scarray_set_value (UfoScarray *scarray,
+                       GValue *value)
+{
+    g_value_set_boxed (value, scarray->array);
```

Comment:
Does this free the old value array?

Reply:
Why should it? This sets the `GValue` (i.e. in `get_property`) not the array.

Reply:
Yeah I confused it because I saw `set`. All good.